subprojects/plugins/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -107,17 +122,25 @@ public DefaultJvmSoftwareComponent(
         assert project.getPlugins().hasPlugin(JavaBasePlugin.class);
 
         this.sourceSet = createMainSourceSet(javaExtension.getSourceSets());
+
+        this.compileJava = tasks.named(sourceSet.getCompileJavaTaskName(), JavaCompile.class);
+        this.jar = registerJarTask(tasks, sourceSet);
+
+        this.implementation = configurations.getByName(sourceSet.getImplementationConfigurationName());
```

Comment:
🤔 Is there a benefit to holding all these configurations as fields rather than retrieving them like this on demand in the getters and as needed?  `createRuntimeElements` already does this.

Reply:
IMO having these as fields helps solidify the fact that the component owns these domain objects. However, the argument of memory usage is a little promising, especially if there are many instances of these per build (why save all of these references if the objects are relatively accessible from the container). 

An advantage to the existing method is that we can also avoid a dependency on the configuration container -- though this container is most likely required for other operations on the component (currently, sources and javadoc jar require it). So, while we have that dependency I think its fine to fetch these objects from the container. 

However, in (distant) future iterations of this component it would be nice if we can avoid holding `Project` or project-derived state after the component has been constructed. Though, currently since that is not feasible, I've updated the code to fetch the configurations from the container on-demand.